allennlp/models/language_model.py

```diff
@@ -0,0 +1,343 @@
+from typing import Dict, List, Tuple, Union
+
+import torch
+import numpy as np
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.models.model import Model
+from allennlp.modules.text_field_embedders import TextFieldEmbedder
+from allennlp.modules.sampled_softmax_loss import SampledSoftmaxLoss
+from allennlp.modules.seq2seq_encoders import Seq2SeqEncoder
+from allennlp.nn.util import get_text_field_mask
+from allennlp.nn import InitializerApplicator
+
+
+class _SoftmaxLoss(torch.nn.Module):
+    """
+    Given some embeddings and some targets, applies a linear layer
```

Comment:
Can you explicitly document here in the constructor that this class does not add a padding ID, and that the input targets to forward should not include the padding id?

Reply:
mind if i do this in another pr? would rather that this changeset not be larger than it has to for the core changes for configurable bidirectionality.

Reply:
looks like it doesn't work :(

Reply:
Oh well.